tests/contrib/django/asgi.py

```diff
@@ -1,4 +1,28 @@
+import os
+
+from channels.auth import AuthMiddlewareStack
+from channels.routing import ProtocolTypeRouter
+from channels.routing import URLRouter
 from django.core.asgi import get_asgi_application
+from django.urls import re_path
+
+
+# Application must be run in the project root to find this settings ex. ddtrace/
+os.environ.setdefault("DJANGO_SETTINGS_MODULE", "tests.contrib.django.django_app.settings")
 
 
 application = get_asgi_application()
+
+channels_application = ProtocolTypeRouter(
+    {
+        "http": AuthMiddlewareStack(
+            URLRouter(
+                [
+                    re_path(r"", application),
```

Comment:
we should include a route here that isn't `application` so we can test that using an asgi application other than Django will work

Reply:
I can use the channels asgi handler for one of the routes (from channels.handler import AsgiHandler)

Reply:
that should be fine, or else just a hello world asgi function should be fine too

Reply:
I added a new application in this commit: abc7ae0

Reply:
I wonder if you need to change the order of these at all... that is is `r"^shutdown-tracer/$"` even necessary / working if everything gets routed to `application` via `r""` ?

Reply:
Yup the order matters and that's what first prevented me from routing to `simple-asgi-app/`. The first case would just sent all requests to the django.core.AsgiHandler

Reply:
Removing the path handlers from the channels URLRouter did the trick. Now i can remove the final `/` from the shutdown-tracer path and allow the django app to handle all the routing 
```
re_path(r"^shutdown-tracer/$", application) 
re_path(r"^error-500/$", application)
```